docs/topics/signals.rst

```diff
@@ -13,9 +13,20 @@ Even though signals provide several arguments, the handlers that catch them
 don't need to accept all of them - the signal dispatching mechanism will only
 deliver the arguments that the handler receives.
 
+Signals underwent a major refactoring with an eye towards speed. Although still
+backwards compatible, receivers now need to have a ``**kwargs`` argument i.e. all
+receivers should now accept a variable keyword args param. We request that you
+make sure all your receivers follow this contract.
```

Comment:
@kmike It's optional as we did not want to break backward compatibility, but those receivers would not fully leverage the full potential of the speedup as `robust_apply` would still be invoked.